lua/ii/disting.lua

```diff
@@ -148,6 +198,46 @@ do return
     , retval = { 'state', u8 }
     , docs = '0-based loop index'
     }
+  , { name = 'dual_parameter'
+    , cmd = 0x5A
+    , args = { { 'side', u8 }
+             , { 'parameter', u8 }
+             }
+    , retval = { 'value', u8 }
+    }
+  , { name = 'dual_parameter_min'
+    , cmd = 0x5B
+    , args = { { 'side', u8 }
+             , { 'parameter', u8 }
+             }
+    , retval = { 'value', u8 }
+    }
+  , { name = 'dual_parameter_max'
+    , cmd = 0x5C
+    , args = { { 'side', u8 }
+             , { 'parameter', u8 }
+             }
+    , retval = { 'value', u8 }
+  , { name = 'dual_algorithm'
+    , cmd = 0x5F
+    , args = { 'side', u8 }
+    , retval = { 'value', u8 }
+    }   }
   }
+, pickle = -- send side/parameter as one byte
+--void pickle( uint8_t* address, uint8_t* data, uint8_t* byte_count );
+[[
+
+uint8_t side = data[1];  // side
+
+if (data[0] >= 0x5A &&
+    data[0] <= 0x5C) {
+  data[1] = (data[1] & 0xF0) | (data[2] & 0xF);
+  data[1] = (data[1] & 0x0F) | ((side & 0xF) << 4);
```

Comment:
i'm out of bit-manipulation practice, but couldn't this just be:
```c
data[1] = ((data[1] & 0xF) << 4) | (data[2] & 0xF);
```
then you could delete the `side` variable.

if there's something trickier going on, perhaps a short inline comment?

Reply:
I'm rusty on manipulation as well... hence a step based approach, but this is much more concise.